libnmstate/varlink/__init__.py

```diff
@@ -0,0 +1,313 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+import os
+import libnmstate
+import argparse
+import logging
+import pkg_resources
+import libnmstate.error as libnmError
+
+
+try:
+    import varlink
+
+    if pkg_resources.get_distribution("varlink").version < "30.2.0":
+        raise libnmError.NmstateDependencyError(
+            "Nmstate-varlink requires varlink package version >= 30.2.0"
+        )
+except ModuleNotFoundError:
+    raise libnmError.NmstateDependencyError(
+        "Varlink dependency package need to be installed."
+    )
+
+OS_ERROR_ADDRESS_ALREADY_IN_USE = 98
+NMSTATE_VARLINK_LOG_FILE = "/var/log/nmstate-varlink.log"
```

Comment:
Saving logs into a file require the handle of log rotation which is too complex for this use case.

The log should return to method caller.